30 création de drivetodock.py

Maxence185 opened the pull request:
Description
-----------
Ajout de la commande Drive to dock 
Avance avec une vitesse qui varie en fonction de l'angle (pitch) du robot
Closes #30 . 

Liste de vérification
---------------------
<!-- Entre les [ ], remplace l'espace par un x lorsque c'est fait. -->
- Writing conventions :
    - [x] English language
    - [x] File names use lowercase without spaces
    - [x] Class names use PascalCase
    - [x] Function and variable names use snake_case
    - [x] Function and command names start with an action verb (get, set, move, start, stop...)
    - [x] Ports respect the naming convention "subsystem" _ "component type" _ "precision"
    - [x] Properties respect the naming convention
    - [x] ntproperty strings are the same as their variables
- Command and subsytem safety :
    - [x] Commands and subsystems inherit from SafeCommand and SafeSubsystem
    - [x] No commands or subsytems have a `setName()` method, unless necessary, because Safe classes automatically set the name
    - [x] Commands have a `addRequirements()` method that includes all the subsystems they use
- [x] J'ai exécuté tout mon code en simulation.

Commit:
création de drivetodock.py

Commit:
Removed setName() for drive to dock

Maxence185 committed:
Merge branch 'main' into 30-avancer-sur-charging-station

etiennebeaulac commented on properties.py:
```diff
@@ -18,6 +18,9 @@ class _Properties:
     drive_deadzone_x = ntproperty("/Properties/drive_deadzone_x", 0.05, writeDefault=False, persistent=persistent)
     drive_deadzone_y = ntproperty("/Properties/drive_deadzone_y", 0.05, writeDefault=False, persistent=persistent)
 
+    drive_to_dock_speed_max = ntproperty("/Propreties/drive_to_dock_speed_max", 0.5, writeDefault=False, persistent=persistent)
+    drive_to_dock_angle_threshold = ntproperty("/Propreties/drive_to_dock_angle_threshold", 0.05, writeDefault=False, persistent=persistent)
+
```
Le fichier `properties` ne devrait plus être utilisé pour les propriétés spécifiques à un fichier. Voir `autoproperty`.

etiennebeaulac commented on commands/drivetodock.py:
```diff
@@ -0,0 +1,25 @@
+        self.moving_average = 1
+
+    def execute(self) -> None:
+        _calculated_speed = (properties.values.drive_to_dock_speed_max * Drivetrain.getPitch() / 180)
+        self.drivetrain.arcadeDrive(_calculated_speed, 0)
+        self.moving_average = (self.moving_average + Drivetrain.getPitch()/180)/2
```
1. Si le pitch initial est de zéro, on ne bougera pas?
2. Il faut utiliser `self.drivetrain` (l'objet, l'instance) et non `Drivetrain` (le nom de la classe, non instancié).
3. Pourquoi est-ce qu'on veut prendre un *moving average* du pitch? Du moment que le docking station "tombe", le pitch diminue et on veut arrêter, non?

Commit:
Merge remote-tracking branch 'origin/main' into 30-avancer-sur-charging-station

Commit:
drive to dock makeover

re-did drive to dock

etiennebeaulac commented on commands/drivetodock.py:
```diff
@@ -0,0 +1,37 @@
+_angle_threshold = 0.5
+_timer_threshold = 2
+
+class DriveToDock(CommandBase):
```
```suggestion
class DriveToDock(SafeCommand):
```

etiennebeaulac commented on commands/drivetodock.py:
```diff
@@ -0,0 +1,37 @@
+        self.timer = wpilib.Timer
+
+    def initialize(self) -> None:
+        self.timer.reset()
```
```suggestion
        self.has_docked = False
        self.timer.stop()
        self.timer.reset()
```

etiennebeaulac commented on commands/drivetodock.py:
```diff
@@ -0,0 +1,37 @@
+        if _pitch <= _angle_threshold and self.has_docked:
+            self.timer.start()
+        else:
+            self.timer.stop()
```
`self.timer.reset()`

etiennebeaulac commented on commands/drivetodock.py:
```diff
@@ -0,0 +1,37 @@
+
+    def execute(self) -> None:
+        _pitch = self.drivetrain.getPitch()
+        _calculated_speed = (_pitch / abs(_pitch)) * (_minimal_drive_speed + abs(_pitch))
```
`_minimal_drive_speed + k_pitch * abs(_pitch)`

Commit:
fixed adis16470 getpitch

Commit:
Merge branch 'main' into 30-avancer-sur-charging-station

Commit:
Responded to suggestions

added weight to the pitch for the _calculated_speed
added timer.reset() before timer.stop()

etiennebeaulac committed:
Quick fix
